Dockerfile

```diff
@@ -91,22 +93,33 @@ RUN install -s bin/* /opt/cni/bin/
 
 
 ### BEGIN CALICO NODE ###
-FROM calico-node-builder AS calico_node
+FROM calico-node-builder-${ARCH} as calico_node
 ARG ARCH
 ARG TAG
 WORKDIR $GOPATH/src/github.com/projectcalico/calico/node
 RUN mkdir -p bin/third-party && go mod download && cp -r ../felix/bpf-gpl/include/libbpf bin/third-party && chmod -R +w bin/third-party
 RUN if [ "${ARCH}" = "amd64" ]; then make -j 16 -C bin/third-party/libbpf/src BUILD_STATIC_ONLY=1; fi
-RUN GO_LDFLAGS="-linkmode=external \
+RUN if [ "${ARCH}" = "amd64" ]; then \  
+    GO_LDFLAGS="-linkmode=external \
     -X github.com/projectcalico/calico/node/pkg/startup.VERSION=${TAG} \
     -X github.com/projectcalico/calico/node/buildinfo.GitRevision=$(git rev-parse HEAD) \
     -X github.com/projectcalico/calico/node/buildinfo.GitVersion=$(git describe --tags --always) \
     -X github.com/projectcalico/calico/node/buildinfo.BuildDate=$(date -u +%FT%T%z)" \
     CGO_LDFLAGS="-L/go/src/github.com/projectcalico/calico/node/bin/third-party/libbpf/src -lbpf -lelf -lz" \
     CGO_CFLAGS="-I/go/src/github.com/projectcalico/calico/node/bin/third-party/libbpf/src" \
-    CGO_ENABLED=1 go build -ldflags "-linkmode=external -extldflags \"-static\"" -gcflags=-trimpath=${GOPATH}/src -o bin/calico-node ./cmd/calico-node
+    CGO_ENABLED=1 go build -ldflags "-linkmode=external -extldflags \"-static\"" -gcflags=-trimpath=${GOPATH}/src -o bin/calico-node ./cmd/calico-node; \
+    else \
+    GO_LDFLAGS="-linkmode=external \
+    -X github.com/projectcalico/node/pkg/startup.VERSION=${TAG} \
+    -X github.com/projectcalico/node/buildinfo.GitRevision=$(git rev-parse HEAD) \
+    -X github.com/projectcalico/node/buildinfo.GitVersion=$(git describe --tags --always) \
+    -X github.com/projectcalico/node/buildinfo.BuildDate=$(date -u +%FT%T%z)" \
+    CGO_ENABLED=0 \
+    CGO_LDFLAGS="" \
+    go build -ldflags "-linkmode=external -extldflags \"-static\"" -gcflags=-trimpath=${GOPATH}/src -o bin/calico-node ./cmd/calico-node; \
+    fi
```

Comment:
Not super important but, what about creating a different `RUN` command for s390x instead of using an `else`? In my opinion, it is nicer to read

Reply:
Yes, I agree.
Done in https://github.com/rancher/image-build-calico/pull/21/commits/70351c99610e9242378603ff061b937b5b3e3791